### PR TITLE
Learn 1338488

### PR DESCRIPTION
**Pull request by chrisdavidmills**
OK, trying this again.

This time, I made sure to only add the template files I modified, with 

git add jinja2/base.html
git add kuma/landing/jinja2/landing/homepage.html

Before committing with git commit -m 'my message'

Does this now work?

### CODE REVIEW

**Comment by chrisdavidmills** [comment]
Dammnit, it still lists the kumascript updates. What am I doing wrong?

**Comment by jwhitlock** [comment]
Thank you for selecting the "allow maintainers to push" checkbox.  Here's what I did:

```
git checkout master  # mozilla/kuma master
git pull --ff-only  # Get latests updates to master, w/o a merge commit
git remote add chrisdavidmills git@github.com:chrisdavidmills/kuma.git  # Add your repo as a remote
git fetch chrisdavidmills  # Get branches
git checkout learn-1338488 # Get the PR branch
git rebase -i master # Start an interactive rebase. Delete the two kumascript commits, save.
git log # Confirm it
git push -f # Force push to chrisdavidmills/learn-1338488, replacing with my version
```

I sense a "Learn GitHub" module for MDN in the future...

File: kuma/landing/jinja2/landing/homepage.html [hunk 1]
```diff
@@ -81,7 +81,7 @@ <h2 class="offscreen">{{ _('Web Technologies') }}</h2>
       <div class="column-feature">
         <h3><i aria-hidden="true" class="icon-graduation-cap"></i><span>{{ _('Learning') }}</span></h3>
```
Comment (jwhitlock) [inline kuma/landing/jinja2/landing/homepage.html]: This should change to "Learn" as well.

**Comment by chrisdavidmills** [comment]
OK, I followed your commands exactly John, let me know if this looks OK. Thanks for all the help here.

And yes, a GitHub module for the learning area would be great. We've actually tried to create one twice, but the author just dropped out and disappeared both times.

I've been using Git(Hub) for years, but have never felt like I know what I am doing. In truth, I have always found it frustrating, confusing, and slightly terrifying. I've never found a tutorial or guide that really helps me in the slightest.

I repeatedly get to a point where I think I understand what's going on, but then (like in this occasion) something always happens that proves I'm clueless. Again.

**Comment by jwhitlock** [comment]
I missed a step 😦.  If you had run ``git fetch --all``, then you would have grabbed my changed branch from GitHub. Instead, ``git checkout learn-1338488`` checked out your local branch with the three extra commits, and now that's the version on GitHub as well.

Please try these commands:

```
git checkout master  # mozilla/kuma master
git pull --ff-only  # Get latests updates to master, w/o a merge commit
git checkout learn-1338488 # Get the PR branch
git rebase -i master # Start an interactive rebase. Delete the two kumascript commits, save.
```

The ``git rebase -i master`` is an interactive rebase, which is the most powerful command in git.  Don't worry.  You'll get something that looks like this (or maybe exactly like this):

```
pick 2da4dedd7 updated kumascript
pick e1e4025fc updated kumascript
pick abdb4f1d0 updating learning -> learn, as per bug 1338488

# Rebase b902a1e64..abdb4f1d0 onto b902a1e64 (3 commands)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

The order is oldest commit at the top, newest commit at the bottom.  The hashes (like ``2da4dedd7``) are the same ones used on Github. I find the Commands documentation to be useful, and I've used them all.  In this case, you want to drop the two "updated kumascript" commits, which you can do by changing ``pick`` to ``drop`` or ``d``, or by removing the lines entirely.

So, remove those two lines and then save the file.  It will "run" the commands, and your branch will be left with just the commit "updating learning -> learn, as per bug 1338488".  The hash will change, because it includes the history of the repo up to that point, but the diff will be the same (``git show``).

OK, back to the commands:
```
git push -f # Force push to chrisdavidmills/learn-1338488, replacing with new version
```

Check on GitHub again, and you should just have the one commit.

**Comment by chrisdavidmills** [comment]
I ran the first four commands of your new message, and am currently staring at the interactive rebase in vim. But I don't have any of the 'pick' lines near the top. My first three lines are:

```
noop

# Rebase ac61f6d..abdb4f1 onto ac61f6d (1 command)
```

Any thoughts?

Also, how can I open up the interactive rebase in a sensible text editor rather than vim? *ducks* ;-) 

I know some people like vim, but I find using it to be an exercise in masochism.

**Comment by jwhitlock** [comment]
What do you get when you run ``git remote -v``? I have many, many lines, but here's the important ones:

```
chrisdavidmills	git@github.com:chrisdavidmills/kuma.git (fetch)
chrisdavidmills	git@github.com:chrisdavidmills/kuma.git (push)
jwhitlock	git@github.com:jwhitlock/kuma.git (fetch)
jwhitlock	git@github.com:jwhitlock/kuma.git (push)
origin	git@github.com:mozilla/kuma.git (push)
origin	git@github.com:mozilla/kuma.git (fetch)
```

**Comment by jwhitlock** [comment]
And then followed by "what is the master branch tracking?": ``git branch -vv | grep master``

```
master                          b902a1e64 [origin/master] Pontoon: Update Thai (th) localization of MDN
```

**Comment by jwhitlock** [comment]
And, to setup an editor that isn't vim: https://help.github.com/articles/associating-text-editors-with-git/

**Comment by chrisdavidmills** [comment]
So, git remote -v gives me

```
mozilla	git@github.com:mozilla/kuma.git (fetch)
mozilla	git@github.com:mozilla/kuma.git (push)
origin	https://github.com/chrisdavidmills/kuma.git (fetch)
origin	https://github.com/chrisdavidmills/kuma.git (push)
```
git branch -vv | grep master gives me

`master        ac61f6d [origin/master: ahead 1] update Learning -> Learn as per bug 1338488`

**Comment by jwhitlock** [comment]
OK, your master branch is not the same as the one on GitHub: https://github.com/mozilla/kuma/commits/master

Let's try this to get it replaced with GitHub's copy:
```
git fetch --all
git checkout master
git reset --hard @{u}
```
Then try [today's commands](https://github.com/mozilla/kuma/pull/4132#issuecomment-279407054) again.

**Comment by jwhitlock** [comment]
Oops, I'm wrong. Let me try again when I'm not in a meeting...

**Comment by jwhitlock** [comment]
I was assuming the remote ``origin`` was the mozilla repo. Bad assumption on my part.  Here's my amended instructions:

```
git fetch --all  # Fetch all remote commits and branches
git checkout learn-1338488 # Get the PR branch
git rebase -i mozilla/master  # Rebase on mozilla/kuma's master, remove the two commits
git push -f  # Replace branch learn-1338488 on chrisdavidmills/kuma
git checkout mozilla/master # Check out mozilla/kuma master in detected head
git push -f origin master # Replace branch master on chrisdavidmills/kuma
```

When you start branches for future PRs, make sure origin/master is up to date with mozilla/master, such as:

```
git fetch --all  # Fetch all remote commits and branches
git checkout mozilla/master  # Checkout master from mozilla/kuma
git push origin/master  # Push changes to master on chrisdavidmills/kuma
git checkout -b my-awesome-branch-12345678  # Create your new PR branch
git push -u origin # and push it to GitHub
```

**Comment by chrisdavidmills** [comment]
OK, I have followed your new set of instructions, and I think I understand what to do better in future. Thanks again for the help John.

**Comment by jwhitlock** [comment]
I'm glad we finally beat git into submission. Thanks for the PR!
